lib/gems/pending/appliance_console/database_configuration.rb

```diff
@@ -119,18 +119,26 @@ def ask_for_database_credentials
       self.database = just_ask("name of the database on #{host}", database) unless local?
       self.username = just_ask("username", username) unless local?
       count = 0
+      empty_count = 0
       loop do
-        count += 1
         password1   = ask_for_password_or_none("database password on #{host}", password)
         # if they took the default, just bail
+        if local? && password1.strip.length == 0
+          raise RuntimeError, "passwords can not be empty" if empty_count > 0
+          empty_count += 1
+          say("\nThe passwords can not be empty.")
+          next
+        end
+ 
         break if (password1 == password)
         password2   = ask_for_password("database password again")
         if password1 == password2
           self.password = password1
           break
-        elsif count > 1 # only reprompt password once
-          raise ArgumentError, "passwords did not match"
+        elsif count > 0 # only reprompt password once
+          raise RuntimeError, "passwords did not match"
```

Comment:
upper `rescue` does not catch `ArgumentError`, and instead of showing main screen (as planned) whole program terminated